business_requirement_sale/wizard/business_requirement_create_sale.py

```diff
@@ -92,16 +92,23 @@ def _onchange_undefined_section(self):
         command = 4 if self.undefined_section else 3
         self.deliverable_ids = [(command, x) for x in deliverables.ids]
 
-    def _prepare_sale_layout_category_vals(self, deliverable, line_vals):
-        return {
-            'name': deliverable.section_id.name,
+    def _prepare_sale_order_line_section_vals(
+            self, section, sequence, order_vals):
+        vals = {
+            'name': section.name,
+            'sequence': sequence,
+            'br_deliverable_section_id': section.id,
+            'display_type': 'line_section',
         }
+        return vals
 
-    def _prepare_sale_order_line_vals(self, deliverable, order_vals):
+    def _prepare_sale_order_line_vals(
+            self, deliverable, sequence, order_vals):
         vals = {
             'name': deliverable.name,
-            'product_id': deliverable.product_id.id,
```

Comment:
Why changing this order in the dict? If you want to put them by alphabetical order, then you must sort `sequence` as well.